app/main.py

```diff
@@ -1,3 +1,49 @@
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: Union["Distance", int]) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union["Distance", int]) -> "Distance":
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: float) -> "Distance":
+        return Distance(self.km * other)
+
+    def __truediv__(self, other: float) -> "Distance":
+        return Distance(round(self.km / other, 2))
+
+    def __lt__(self, other: float) -> bool:
+        return self.__lt__(other.km) if isinstance(other, Distance) \
+            else self.km < other
+
+    def __gt__(self, other: float) -> bool:
+        return self.__gt__(other.km) if isinstance(other, Distance) \
+            else self.km > other
+
+    def __eq__(self, other: float) -> bool:
+        if isinstance(other, Distance):
+            return self.km == other.km
+        return self.km == other
+
+    def __le__(self, other: float) -> bool:
+        return not self.__gt__(other)
+
+    def __ge__(self, other: float) -> bool:
+        return not self.__lt__(other)
```

Comment:
u are close)
but it can be written simpler
use operand `<` instead of `self.__lt__(other)` etc

Reply:
kk  😶‍🌫️